python/ray/train/gbdt_trainer.py

```diff
@@ -24,18 +24,37 @@ def _convert_scaling_config_to_ray_params(
     ray_params_cls: Type["xgboost_ray.RayParams"],
     default_ray_params: Optional[Dict[str, Any]] = None,
 ) -> "xgboost_ray.RayParams":
-    default_ray_params = default_ray_params or {}
-    resources_per_worker = scaling_config.additional_resources_per_worker
-    num_workers = scaling_config.num_workers
-    cpus_per_worker = scaling_config.num_cpus_per_worker
-    gpus_per_worker = scaling_config.num_gpus_per_worker
+    ray_params_kwargs = default_ray_params.copy() or {}
+
+    resources = scaling_config.resources_per_worker or {}
+
+    cpus_per_actor = resources.pop("CPU", 0)
+    if not cpus_per_actor:
+        cpus_per_actor = ray_params_kwargs.get("cpus_per_actor", 0)
+
+    gpus_per_actor = resources.pop("GPU", 0)
```

Comment:
is there a reason not to use `scaling_config.num_cpus_per_worker` and `scaling_config.num_gpus_per_worker`?

Reply:
Because e.g. the scaling_config.num_cpus_per_worker always defaults to 1. We can't distinguish between something the user set (if they specifically request CPUs=1) and the implicit default.

Reply:
I see, but will this work for the GPU case? If I set `use_gpu=True` in my scaling config, `resources` will still be None, and then in this logic here `gpus_per_actor` will be 0, right?

Reply:
Ah that's right. Let me fix that